lib/miq_pglogical/miq_pglogical_subscriber.rb

```diff
@@ -0,0 +1,93 @@
+require 'miq-password'
+
+class MiqPglogicalSubscriber < MiqPglogical
+  # Configures the subscriber node
+  # This should be called when pglogical is being used for the first time
+  def configure_subscriber
+    return if subscriber?
+    pglogical.enable
+    create_node unless node?
+  end
+
+  # Destroys all subscriptions and all data associated with them
+  # This also destroys the subscriber node
+  def destroy_subscriber
+    active_subscription_names.each { |name| pglogical.subscription_drop(name) }
+    MiqRegion.where.not(:region => ApplicationRecord.my_region_number).pluck(:region).each do |r|
+      MiqRegion.destroy_region(@connection, r)
+    end
+    c = MiqServer.my_server.get_config
+    c.config.store_path(*SETTINGS_PATH, :subscriptions, [])
+    c.save
+    drop_node
+  end
+
+  # Lists the subscriptions in the replication configuration
+  # @return Array<Hash<Symbol, String>> the list of provider database connection information
+  #   This hash should contain the keys:
+  #     :dbname, :user, :password, :host, :port
+  def configured_subscriptions
+    name_configured_subscriptions
+    MiqServer.my_server.get_config.config.fetch_path(*SETTINGS_PATH, :subscriptions) || []
+  end
+
+  # Brings the running subscriptions into compliance with the replication configuration
+  def refresh_subscriptions
+    added_subscription_conf.each { |conf| create_subscription(conf) }
+    removed_subscription_names.each do |name|
+      pglogical.subscription_drop(name)
+      MiqRegion.destroy_region(@connection, region_for_subscription_name(name))
+    end
+  end
+
+  private
+
+  def name_configured_subscriptions
+    c = MiqServer.my_server.get_config
+    return unless (subscriptions = c.config.fetch_path(*SETTINGS_PATH, :subscriptions))
+    subscriptions.each do |s|
+      next if s[:name]
+      s[:name] = "subscription_#{s[:host].gsub(/\.|-/, "_")}"
+    end
+    c.config.store_path(*SETTINGS_PATH, :subscriptions, subscriptions)
+    c.save
+  end
+
+  def create_subscription(conf)
+    pglogical.subscription_create(conf[:name], dsn_from_conf_hash(conf), [REPLICATION_SET_NAME], false)
+  end
+
+  def added_subscription_conf
+    configured   = configured_subscriptions
+    active_names = active_subscription_names
+    configured.each_with_object([]) do |config, new_conf|
+      new_conf << config unless active_names.include?(config[:name])
```

Comment:
This one is a bit odd because configured is an array of hashes and I want to get the full hashes whose names are not in active_names.

So I want to end up with another array of hashes.
Probably this?

```ruby
configured.reject do |config|
  active_names.include?(config[:name])
end
```